src/software/simulated_tests/corner_kick_play_test.cpp

```diff
@@ -0,0 +1,91 @@
+#include "software/ai/hl/stp/play/corner_kick_play.h"
+
+#include <gtest/gtest.h>
+
+#include "software/simulated_tests/simulated_test_fixture.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class CornerKickPlayTest : public SimulatedTestFixture
+{
+};
+
+TEST_F(CornerKickPlayTest, test_corner_kick_play)
+{
+    setBallState(BallState(Point(4.5, -3), Vector(0, 0)));
+    addFriendlyRobots({
+        RobotStateWithId{
+            .id          = 0,
+            .robot_state = RobotState(Point(-3, 2.5), Vector(0, 0), Angle::zero(),
+                                      AngularVelocity::zero())},
+        RobotStateWithId{
+            .id          = 1,
+            .robot_state = RobotState(Point(-3, 1.5), Vector(0, 0), Angle::zero(),
+                                      AngularVelocity::zero())},
+        RobotStateWithId{
+            .id          = 2,
+            .robot_state = RobotState(Point(-3, 0.5), Vector(0, 0), Angle::zero(),
+                                      AngularVelocity::zero())},
+        RobotStateWithId{
+            .id          = 3,
+            .robot_state = RobotState(Point(-3, -0.5), Vector(0, 0), Angle::zero(),
+                                      AngularVelocity::zero())},
+        RobotStateWithId{
+            .id          = 4,
+            .robot_state = RobotState(Point(-3, -1.5), Vector(0, 0), Angle::zero(),
+                                      AngularVelocity::zero())},
+        RobotStateWithId{
+            .id          = 5,
+            .robot_state = RobotState(Point(4.6, -3.1), Vector(0, 0), Angle::zero(),
+                                      AngularVelocity::zero())},
+    });
```

Comment:
probably worth adding a helper in the `simulated_test_fixture`  to create a vector of `RobotStateWithId` from a vector of `Point`s

Reply:
done